modules/gltf/src/pbr/parse-pbr-material.ts

```diff
@@ -257,39 +243,98 @@ function addTexture(
   define: string,
   parsedMaterial: ParsedPBRMaterial
 ): void {
-  const parameters = gltfTexture?.texture?.sampler?.parameters || {};
-
   const image = gltfTexture.texture.source.image;
   let textureOptions;
-  let specialTextureParameters = {};
+
   if (image.compressed) {
     textureOptions = image;
-    specialTextureParameters = {
-      [GLEnum.TEXTURE_MIN_FILTER]:
-        image.data.length > 1 ? GLEnum.LINEAR_MIPMAP_NEAREST : GLEnum.LINEAR
-    };
   } else {
     // Texture2D accepts a promise that returns an image as data (Async Textures)
     textureOptions = {data: image};
   }
 
+  const gltfSampler = {
+    wrapS: 10497, // default REPEAT S (U) wrapping mode.
+    wrapT: 10497, // default REPEAT T (V) wrapping mode.
+    ...gltfTexture?.texture?.sampler
+  } as any;
+
   const texture: Texture = device.createTexture({
     id: gltfTexture.uniformName || gltfTexture.id,
-    // TODO: FIX THIS
-    parameters: {
-      ...parameters,
-      ...specialTextureParameters
-    },
-    pixelStore: {
-      [GLEnum.UNPACK_FLIP_Y_WEBGL]: false
-    },
+    sampler: convertSampler(gltfSampler),
     ...textureOptions
   });
+
   parsedMaterial.bindings[uniformName] = texture;
   if (define) parsedMaterial.defines[define] = true;
   parsedMaterial.generatedTextures.push(texture);
 }
 
+export function convertSampler(gltfSampler: any): SamplerProps {
```

Comment:
Let's create a separate files `convert-webgl.ts` (and `convert-webg.spec.ts` with maching tests) for these utilities. That keeps the other code cleaner and gives us the option to easily move these to e.g. loaders.gl later.

Also note that `@luma.gl/constants` does export some types in addition to the constants so you should be able to avoid `any` here:

```suggestion
import {type GLSamplerParameters} from '@luma.gl/constants`;
export function convertSampler(gltfSampler: GLSamplerParameters): SamplerProps {
```

Reply:
The type GLSamplerParameters does not match the gltf.

Reply:
> The type GLSamplerParameters does not match the gltf.

If there is nothing that works the type out the possible values yourself so we can drop the "any"?

Or at least use "number"?